packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.controller.js

```diff
@@ -0,0 +1,129 @@
+import filter from 'lodash/filter';
+import find from 'lodash/find';
+import get from 'lodash/get';
+import map from 'lodash/map';
+
+export default class PrivateSqlActivationController {
+  /* @ngInject */
+  constructor($q, $translate, OrderService, OvhApiHostingWeb) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.OrderService = OrderService;
+    this.OvhApiHostingWeb = OvhApiHostingWeb;
+  }
+
+  $onInit() {
+    this.loading = {
+      hosting: false,
+      checkout: false,
+    };
+    this.error = {
+      hosting: null,
+      checkout: null,
+    };
+
+    this.autoPayWithPreferredPaymentMethod = false;
+    this.acceptContracts = false;
+
+    if (this.hosting) {
+      this.onHostingChange();
+    }
+  }
+
+  onHostingChange() {
+    this.loading.hosting = true;
+    this.error.hosting = null;
+    this.option = undefined;
+    return this.$q.all([
+      this.fetchDataCenter(),
+      this.fetchOptions(),
+    ]).catch((error) => {
+      this.error.hosting = error;
+    }).finally(() => {
+      this.loading.hosting = false;
+    });
+  }
+
+  fetchOptions() {
+    return this.OrderService
+      .getProductServiceOptions('webHosting', this.hosting)
+      .then((options) => filter(options, (option) => option.planCode.startsWith('private-sql')))
+      .then((options) => {
+        this.options = map(options, (option) => {
+          const price = get(find(option.prices, { pricingMode: 'default' }), 'price.text');
+          const planLabel = this.$translate.instant(`privatesql_activation_option_${option.planCode}`);
+          return {
+            value: option,
+            label: `${planLabel} - ${price}`,
+          };
+        });
+        return this.options;
+      });
+  }
+
+  fetchDataCenter() {
+    return this.OvhApiHostingWeb.v6().get({
+      serviceName: this.hosting,
+    }).$promise
+      .then(({ datacenter }) => {
+        this.datacenter = datacenter;
+        return datacenter;
+      });
+  }
+
+  fetchCheckoutInformations() {
+    const price = find(this.option.value.prices, { pricingMode: 'default' });
+    this.loading.checkout = true;
+    this.error.checkout = null;
+    return this.OrderService.createNewCart(this.me.ovhSubsidiary)
```

Comment:
```suggestion
    return this.wucOrderService.createNewCart(this.me.ovhSubsidiary)
```
as it has been merged that way this will avoid having to go through this again

Reply:
FYI : @marie-j / @frenautvh, it is `this.WucOrderCartService`